### PR TITLE
Setting up prettier

### PR DESCRIPTION
**Pull request by moufmouf**
In order not to wreak havoc on all the open PRs, Prettier will be added slowly,
on every changed file, thanks to Husky (that sets up a precommit hook) and lint-staged, for linting only staged files.

This PR also contains a CONTRIBUTING.md guide explaining how to set up the Prettier pre-commit hook.

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+You can ask us on [Discord](https://discord.gg/YGtngdh9gt) or in the [GitHub issues](https://github.com/thecodingmachine/workadventure/issues).
+
+### Linting your code
+
+Before committing, be sure to install the "Prettier" precommit hook that will reformat your code to our coding style.
+
+In order to enable the "Prettier" precommit hook, at the root of the project, run:
+
+```console
+$ yarn run install
```
Comment (jonnytest1) [inline CONTRIBUTING.md:45]: its actually yarn install

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+$ yarn run prepare
```
Comment (jonnytest1) [inline CONTRIBUTING.md:46]: its called pre-commit

File: back/.prettierrc.json [hunk 3]
```diff
@@ -0,0 +1,3 @@
+{
+  "tabWidth": 4
```
Comment (jonnytest1) [inline back/.prettierrc.json:2]: yeah we gonna need loads more rules :D

Comment (jonnytest1) [reply back/.prettierrc.json:2]: quotestyle 
indent rules 
newline rules

member ordering 
import ordering
my recommendations

Comment (moufmouf) [reply back/.prettierrc.json:2]: Do you have a sample .prettierrc.json file I could reuse? Or do you want to open a PR with it?

Comment (jonnytest1) [reply back/.prettierrc.json:2]: sry i mostly use eslint or tslint and the one project i use prettier is using the js configuration

**Pull request merged**